### PR TITLE
Checking for TaggedValue Component

### PR DESCRIPTION
**Pull request by amolskh**
I have added style to tagged Value data structure instead of taggedValueList
You can review and see if basic approach and let me know

### CODE REVIEW

File: components/TaggedValue/TaggedValue.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require('./TaggedValue.scss')
+
+import React from 'react'
+import classNames from 'classnames'
+
+const TaggedValue = ({title,subText,style,count}) => {  
+  const taggedValueClass = classNames(
+    'TaggedValue',
+    { [`${style}`]: style,
```
Comment (vikasrohit) [inline components/TaggedValue/TaggedValue.jsx]: Did you try `[style] : true`? It should work. And you can set a default value, which you can define in TaggedValue.scss, for style property. Or Am I missing something here?

Comment (amolskh) [inline components/TaggedValue/TaggedValue.jsx]: If style is not supplied using true result in 'undefined' string in html

So used style

Comment (vikasrohit) [inline components/TaggedValue/TaggedValue.jsx]: What if we have a default value for that and use that default class to style the element in a default color?

Comment (amolskh) [inline components/TaggedValue/TaggedValue.jsx]: thats already taken care by TaggedValue class only if want to customize you provide style

**Comment by vikasrohit** [comment]
Let me rephrase it again. What I am saying is that the css rule where you are setting the background color of the label is `tagged-value-bg` right now and you are providing the custom class at component level i.e. parent of the `tagged-value-bg`. Now, if you set a default value, lets say `default`, to the `style` property, you can remove the css rule of background color from `tagged-value-bg` class and instead use following:
`
.TaggedValue {
  &.default {
    .tagged-value-bg {
      background-color: $accent;
    }
  }
}
`
Does this makes sense?

File: components/TaggedValue/TaggedValue.jsx (hunk 1 above)
Comment (vikasrohit) [inline components/TaggedValue/TaggedValue.jsx]: Let me rephrase it again. What I am saying is that the css rule where you are setting the background color of the label is `tagged-value-bg` right now and you are providing the custom class at component level i.e. parent of the `tagged-value-bg`. Now, if you set a default value, lets say `default`, to the `style` property, you can remove the css rule of background color from `tagged-value-bg` class and instead use following:
`
.TaggedValue {
  &.default {
    .tagged-value-bg {
      background-color: $accent;
    }
  }
}
`
Does this makes sense?

**Comment by vikasrohit** [comment]
This branch also has conflicts which needed to solved on your local machine.

**Comment by amolskh** [comment]
Ok but I am just wondering what is advantage of doing same

I mean whats the objective of doing this way

Regards

**Comment by vikasrohit** [comment]
@amolskh Can you please move the [comment](https://github.com/appirio-tech/react-components/pull/68#issuecomment-204488643) as line note after my last reply for that section? That would make the discussion more readable, other wise we will lose track of why we did the change.

**Comment by amolskh** [comment]
Oh Sorry email replied .. Moved

File: components/TaggedValue/TaggedValue.jsx (hunk 1 above)
Comment (amolskh) [inline components/TaggedValue/TaggedValue.jsx]: Ok but I am just wondering what is advantage of doing same
I mean whats the objective of doing this way

Comment (vikasrohit) [inline components/TaggedValue/TaggedValue.jsx]: Benefit is that we have relatively simple code `[style] : true` instead of  `[``${style}``]: style` and it would be more like themes to use class for default style as well. `tagged-value-bg` class would contain rules that are rare to change for the component, mostly we would be changing background and font colors which we can move to class. This way we can support more themes in the component itself, the caller may not need to define the background and font colors. They can simply specify the supported theme name.

Comment (amolskh) [inline components/TaggedValue/TaggedValue.jsx]: OK I dont have issue changing to it . Actually initially it was same what you are saying
So now if in JSX tag the attribute style is not provided this is html generated(note style variable is initialised to undefined)
class = "TaggedValue undefined " or if you add default class class = "TaggedValue undefined default" 

since we have set style to true always... If required I cn go ahead and change it

Comment (vikasrohit) [inline components/TaggedValue/TaggedValue.jsx]: Ok. Leave it for now. I would check it later.

**Comment by vikasrohit** [comment]
@amolskh Just a reminder, submission is closing in few minutes. Please submit the PR with conflicts resolved. We can fix other issues in review phase.

**Comment by amolskh** [comment]
After merging the latest changes I am unable to deploy it locally Getting error 
Cannot find module "React"   Any idea if I need to do anything else

**Comment by vikasrohit** [comment]
Did you try after removing the node_modules and doing `npm install` again?

**Comment by amolskh** [comment]
OK i have done the fixes and checked in the code..
What do you mean by point 6? its not clear to me
I have used different colors to show that you can change default styles from example.scss

**Comment by vikasrohit** [comment]
Point 6 is just for cross checking how scroll bar behaves on mobile devices. Ideally we want it not to be visible on mobile devices. Anyway, we can leave it for now. We would handle it internally.

File: components/TaggedValue/TaggedValue.scss [hunk 2]
```diff
@@ -23,7 +23,7 @@ $accent: #59A7FF;
     color:  $gray;
     padding: 1px 7px 2px 7px;
```
Comment (vikasrohit) [inline components/TaggedValue/TaggedValue.scss]: Padding is set twice. Check line number 13.

**Comment by vikasrohit** [comment]
I think the subText is not vertically middle aligned yet with the label. Can you please once again?

**Comment by vikasrohit** [comment]
Merged. Will post final fixes soon.
